### ISSUE TITLE
Compatibility with 3.6.x

### ISSUE THREAD

**Issue by infabo** [wirecard/magento-wcp#1]
Is there any special reason, why v4.x dropped compatibility with v3?
It is not documented anywhere (changelog, etc.).

For example:
v3: payment/wirecard_checkout_page_paypal
v4: payment/wirecard_checkoutpage_paypal

Uninstalling the old version results in not beeing able to view old orders with payments from v3 anymore.
Only workaround to his I have found, is to keep v3 installed too.
But this looks like really a lame solution.

Any thoughts on this?
Thanks!

**Comment by mbretter** [comment]
Hi,

the v4 version is a complete rewrite (split up into to several plugins for checkout page and checkout seamless), so there is no upgrade possible from v3, therefore we decided to use different namespaces, avoid conflicts, if both plugins v3 and v4 are installed.

Your workaround looks ok and is the preferred solutions, if you would like to view payments of the old orders.

cheers

**Comment by infabo** [comment]
Ok, thanks for your answer. As 3.6.4 contains a major-bug (IMHO) - there is no way around upgrading to v4.

v3.6.4:

app/code/community/Phoenix/WirecardCheckoutPage/Model/InvoiceB2b.php

Lines 98 - 99
```
        if (strlen($billingAddress->getCompany()))
            return true; //FAIIIIILLLLLLLLL
```

This enables Invoice B2B method, if billing-company is filled out. Mind blown!

**Comment by Trimnosh** [comment]
Hi!

The described bug was fixed in version 3.6.7. You can get this version from Wirecard support. 

Since the payment methods are different in versions 3.6 and 4.0, the result of removing the plugin would be the removal of payment methods, which is not that easy in Magento, therefore keeping the old plugin is necessary to display the orders made with version 3.6.

If you want to upgrade to v4.0 and not see any of the payment methods of v3.6, you can make them invisible in the file `app/code/community/Phoenix/WirecardCheckoutPage/etc/system.xml` by setting `show_in_default`, `show_in_website` and `show_in_store` to 0.
